src/services/marp_service.py

```
@@ -0,0 +1,65 @@
+    def generate_pptx(self, output_filename="slides.pptx", theme=None):
+        return self._generate(self.OutputType.PPTX, output_filename, theme=theme)
+
+    def _generate(self, output_type, output_filename, theme=None):
+        if not self.output_dir:
+            raise ValueError("Output directory must be set for generation.")
+        output_path = os.path.join(self.output_dir, output_filename)
+        command = ["marp", self.slides_path, "-o", output_path]
+        if theme:
+            command.extend(["--theme", theme])
+        try:
+            result = subprocess.run(
```

Ruff (0.12.2)

`subprocess` call: check for execution of untrusted input

(S603)